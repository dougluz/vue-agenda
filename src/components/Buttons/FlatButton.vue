<template>
  <button class="flat-button">
    <slot> Cancelar </slot>
  </button>
</template>

<style scoped>
.flat-button {
  @apply text-sm font-medium text-light-red rounded;
  padding: 0.75rem 1.375rem 0.75rem 1rem;
}
</style>
