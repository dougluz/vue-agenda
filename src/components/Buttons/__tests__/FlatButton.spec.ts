import { mount } from "@vue/test-utils";
import { describe, it, expect } from "vitest";
import FlatButton from "../FlatButton.vue";

describe("FlatButton", () => {
  describe("when rendering the default version", () => {
    it("should match the snapshot", () => {
      const wrapper = mount(FlatButton);

      expect(wrapper.html()).toMatchSnapshot();
    });

    it("should show the default text", () => {
      const wrapper = mount(FlatButton);

      expect(wrapper.text()).toBe("Cancelar");
    });
  });
  describe("when passing a custom text in the slot", () => {
    it("should show the custom text", () => {
      const wrapper = mount(FlatButton, {
        slots: {
          default: "Custom text",
        },
      });

      expect(wrapper.text()).toBe("Custom text");
    });
  });
});
