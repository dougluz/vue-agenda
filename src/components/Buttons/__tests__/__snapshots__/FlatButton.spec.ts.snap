// Vitest Snapshot v1

exports[`FlatButton > when rendering the default version > should match the snapshot 1`] = `"<button class=\\"flat-button\\" data-v-12f2c869=\\"\\"> Cancelar </button>"`;
